cg/apps/demultiplex/sample_sheet/sample_models.py

```diff
@@ -0,0 +1,220 @@
+import logging
+from abc import abstractmethod
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from cg.apps.demultiplex.sample_sheet.index import (
+    MINIMUM_HAMMING_DISTANCE,
+    get_hamming_distance_for_indexes,
+    get_reverse_complement_dna_seq,
+    is_dual_index,
+    is_padding_needed,
+    pad_index_one,
+    pad_index_two,
+)
+from cg.apps.demultiplex.sample_sheet.validators import SampleId
+from cg.constants.constants import GenomeVersion
+from cg.constants.demultiplexing import (
+    CUSTOM_INDEX_TAIL,
+    SampleSheetBcl2FastqSections,
+    SampleSheetBCLConvertSections,
+)
+from cg.constants.symbols import DASH, EMPTY_STRING, LOWER_CASE_NA
+from cg.exc import SampleSheetError
+from cg.models.demultiplex.run_parameters import RunParameters
+
+LOG = logging.getLogger(__name__)
+
+
+class FlowCellSample(BaseModel):
+    """Base class for flow cell samples."""
+
+    lane: int
+    sample_id: SampleId
+    index: str
+    index2: str = EMPTY_STRING
+    model_config = ConfigDict(populate_by_name=True, extra="ignore")
```

Comment:
```suggestion
    model_config = ConfigDict(populate_by_name=True)
```
extra is set to ignore by default. This change won't affect behaviour though.

Reply:
implemented